.github/workflows/scripts/code_coverage_disable_check.py

```
@@ -0,0 +1,177 @@
+            # If it's a directory, recursively walk through the files in it
+            for root, _, files in os.walk(item):
+                if "node_modules" in root:
+                    continue
+                for file_name in files:
+                    if (
+                        file_name.endswith(".tsx")
+                        or file_name.endswith(".ts")
+                        and not file_name.endswith(".test.tsx")
+                        and not file_name.endswith(".test.ts")
+                        and not file_name.endswith(".spec.tsx")
+                        and not file_name.endswith(".spec.ts")
```

Ruff (0.12.2)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)